WordSearchTest.java

```diff
@@ -0,0 +1,22 @@
+import org.junit.Test;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class WordSearchTest {
+    @Test
+    public void testFindAllWords() {
+        List<String> myDictionary = Arrays.asList("CAR", "CARD", "CART", "CAT");
+        List<String> prefixes = Arrays.asList("C", "CA", "CAR", "CARD", "CART", "CAT");

```

Comment:
I know that Dictionary class is out of scope of the assignment, but prefixes could be trivially generated from the list of words, saving you a bit of error-prone code.